code/src/components/GameScreen.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import styled from 'styled-components/macro'
+
+import games, { generateQuestion } from '../reducers/games'
+
+const GameScreen = () => {
+  const description = useSelector(store => store.games.description)
+  const moves = useSelector(store => store.games.moves)
+  const history = useSelector(store => store.games.history)
+  const coordinates = useSelector(store => store.games.coordinates)
+  
+  const dispatch = useDispatch()
+
+  const onChooseDirection = (event) => {
+    dispatch(games.actions.setDirection(event.target.value))
+    dispatch(generateQuestion())
+  }
+  
+  const onMovesRevert = () => {
+    dispatch(games.actions.setPreviousMove())
+  }
+
+  const onRestartButton = () => {
+    dispatch(games.actions.setUserName(''))
+    window.location.reload()
+  }
+
+  return(
+    <Background>
+      <ButtonContainer>
+        {history.length
+          ? <BackButton
+          onClick={onMovesRevert}>
+            GO BACK
+          </BackButton>
+          : <></>
+        }
+        <RestartButton
+          onClick={onRestartButton}>
+          RESTART
+        </RestartButton>
+      </ButtonContainer>
+      <OuterContainer>
+        <Text>{description}</Text>
+        {moves && moves.map(move => (
+          <InnerContainer key={move.description}>
+            <Text>{move.description}</Text>
+            <Button 
+              value={move.direction}
+              onClick={(event) => onChooseDirection(event)}>
+                Go {move.direction}
+            </Button>
+          </InnerContainer>
+          ))
+          }  
+          {coordinates === '1,3' && (
+            <Diamond />
```

Comment:
Not sure if this is wrong, but it looks like Diamond is a component.  Or is it just av div with a missing closing tag?